.gitignore

```diff
@@ -3,7 +3,7 @@ bench_ecmult
 bench_internal
 tests
 exhaustive_tests
-gen_context
+gen_ecmult_gen_static_prec_table
```

Comment:
It's too bad that `gen` stands for both "generate" (the static table) and "generator" (of a group) 😅 , but it's probably not worth changing, especially not as part of this PR.

Reply:
Yeah, this confused me a lot when starting to work on the library a few years back. And `gen_context` should have been called `gen_gen_context` at least, even though that's ugly.

I tried to avoid this discussion in this PR (only renaming thins I'm touching anyway) and it may be worth unifying the names in the follow-up PR that switches to linking instead of including (e.g., I intentionally used "create" in a function name instead of "generate") Then we anyway need to touch both "generate" things again, so this would be a good opportunity.